Assets/Scripts/Actor/BossAttack.cs

```diff
@@ -0,0 +1,54 @@
+﻿using BulletManagement;
+using JetBrains.Annotations;
+using SlipTime;
+using UnityEngine;
+using System.Collections.Generic;
+
+namespace Actor
+{
+    public class BossAttack: MonoBehaviour
+    {
+        // How long the attack lasts if the player times it out
+        public float durationSeconds;
+
+        // How much damage the player must deal to cancel the attack
+        public int health;
+
+        // For special attacks, define an edgy name
+        [CanBeNull] public string attackName;
+        
+        // For special attacks, define a capture bonus
+        public int scoreBonus = 0;
+
+        // A list of the emitters associated with this pattern
+        public GameObject[] emitterPrefabs;
+        
+        // A list of the emitters saved for destruction later
+        private List<GameObject> emitterInstances = new List<GameObject>();
+
+        // A list of the xy coordinates for each emitter
+        public int[] xCoords, yCoords;
```

Comment:
I hate doing it this way, but the Unity editor won't display any kind of key-value type, so I would have to go and define a custom serialized class for whatever built-in type I used, which seems silly. This gets the job done without too much inconvenience.

Reply:
Actually there is a better way to do this. Add sub-gameobjects to each of your BossAttack gameobjects before you save the prefab. Then access all sub-gameobjects in the Start script. That way you can visually move the emitters before you actually save the emitters AND you get transformations as well (if parent moves, then the children as well.)